common/model/cpptransaction.cc

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "common/model/cpptransaction.h"
+#include "common/trinary/trit_array.h"
+
+namespace iota {
+namespace model {
+Transaction::Transaction() { iota_transaction = transaction_new(); }
+
+Transaction::Transaction(std::string trytes) {
+  iota_transaction = transaction_new();
+  size_t to_len = trytes.size() * 3;
+  size_t flex_len = flex_trits_num_for_trits(to_len);
+  flex_trit_t trits[flex_len];
+  tryte_to_flex_trit(trits, to_len, (const tryte_t *)trytes.data(),
+                     trytes.size(), trytes.size());
+  transaction_deserialize_from_trits(iota_transaction, trits);
+}
+
+Transaction::~Transaction() { transaction_free(iota_transaction); }
+
+std::string Transaction::get_signature(void) {
+  auto trits = transaction_signature(iota_transaction);
+  tryte_t trytes[2187];
+  flex_trit_to_tryte(trytes, 2187, trits, 6561, 6561);
+  return std::string(reinterpret_cast<char *>(trytes), 2187);
+}
+
+// Set the transaction signature (copy argument)
+void Transaction::set_signature(std::string trytes) {
+  size_t flex_len = flex_trits_num_for_trits(6561);
+  flex_trit_t trits[flex_len];
+  tryte_to_flex_trit(trits, 6561, (const tryte_t *)trytes.data(), 2187, 2187);
+  transaction_set_signature(iota_transaction, trits);
+}
+
+// Get the transaction message
+std::string Transaction::get_message(void) {
+  auto trits = transaction_message(iota_transaction);
+  tryte_t trytes[2187];
+  flex_trit_to_tryte(trytes, 2187, trits, 6561, 6561);
+  return std::string(reinterpret_cast<char *>(trytes), 2187);
+}
+
+// Set the transaction message (copy argument)
+void Transaction::set_message(std::string trytes) {
+  size_t flex_len = flex_trits_num_for_trits(6561);
+  flex_trit_t trits[flex_len];
+  tryte_to_flex_trit(trits, 6561, (const tryte_t *)trytes.data(), 2187, 2187);
+  transaction_set_message(iota_transaction, trits);
+}
+
+// Get the transaction address
+std::string Transaction::get_address(void) {
+  auto trits = transaction_address(iota_transaction);
+  tryte_t trytes[81];
+  flex_trit_to_tryte(trytes, 81, trits, 243, 243);
+  return std::string(reinterpret_cast<char *>(trytes), 81);
+}
+
+// Set the transaction address (copy argument)
+void Transaction::set_address(std::string trytes) {
+  size_t flex_len = flex_trits_num_for_trits(243);
+  flex_trit_t trits[flex_len];
+  tryte_to_flex_trit(trits, 243, (const tryte_t *)trytes.data(), 81, 81);
```

Comment:
Instead of allocating `flex_trit_t` on the stack, we should write straight to the element on the `iota_transaction` instance. This saves quite a bit of memory!

Reply:
Same goes for all the other methods here as well